Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -762,11 +773,9 @@ class CombinatorialKalmanFilter {
             nBranchesOnSurface = 0;
           }
         }
-        if (surface->surfaceMaterial() != nullptr) {
```

Comment:
This check is not needed?

Reply:
this check was redundant as we also check for that in the interaction method